### PR TITLE
Add support for DMA

### PR DESCRIPTION
**Pull request by khrs**
Port of STM32 DMA driver to GD32VF world. All previous @Disasm notes included.

### CODE REVIEW

**Comment by aw** [comment]
This is awesome, thank you!

File: src/dma.rs [hunk 1]
```diff
@@ -0,0 +1,584 @@
...
+            $PADDRX:ident,
+            $MADDRX:ident,
+            $ctlX:ident,
+            $cntX:ident,
+            $paddrX:ident,
+            $maddrX:ident,
+            $errifX:ident,
+            $htfifX:ident,
+            $ftfifX:ident,
+            $errifcX:ident,
```
Comment (Disasm) [inline src/dma.rs:134]: `$errifX` and `$errifcX` are not used in the code. Maybe they should be removed?

**Review by Disasm — COMMENTED** [review]
Thank you! Sorry for the delay.
Overall looks good, but I'd make functions `set_peripheral_address` and `set_memory_address` unsafe.

**Comment by Disasm** [comment]
Going to merge this and fix problems soon

**Pull request merged**